### PR TITLE
pin fiona in oldest

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR
* pins `fiona<1.10` in the the `310-oldest` CI environment
* cleans up & organizes CI environments

### CODE REVIEW

**Comment by martinfleis** [comment]
Somehow it is not strictly needed here as our tests in here do not invoke that incompatible code. 

Also, no clue what that error is.

**Comment by jGaboardi** [comment]
> Somehow it is not strictly needed here as our tests in here do not invoke that incompatible code.

Good to have for conformity though, no?

> Also, no clue what that error is.

That was due to a `yaml` syntax error

File: ci/310-latest.yaml [hunk 1]
```diff
@@ -3,59 +3,50 @@ channels:
...
+  - spopt
+  - spreg
+  - spvcm
+  - tobler>=0.2.1
+  # external & optiopnal
   - bokeh
-  - scikit-learn
+  - deprecated
+  - dill
+  - fiona
```
Comment (martinfleis) [inline ci/310-latest.yaml:28]: Do we need Fiona at all?

Comment (martinfleis) [reply ci/310-latest.yaml:28]: I mean, in the latest. We need that pin in the oldest.

Comment (jGaboardi) [reply ci/310-latest.yaml:28]: suppose not?

**Review by gegen07 — APPROVED** [review]
Why we should pin fiona to 1.10 in oldest? Maybe I am missing something important.

**Comment by jGaboardi** [comment]
> Why we should pin fiona to 1.10 in oldest? Maybe I am missing something important.

See [here](https://github.com/pysal/momepy/pull/654). Has to do with older `geopandas` mixing with latest `fiona`

**Comment by martinfleis** [comment]
@gegen07 we're pinning old geopandas which is not compatible with the latest Fiona. If we don't pin Fiona, we hit an error on missing `fiona.path` module. It is more important in CI envs where this actually happens than here.

**Review by knaaptime — APPROVED** [review]

**Pull request merged**